ui/packages/catalog-ui-search/src/main/webapp/component/sharing/sharing.view.js

```diff
@@ -0,0 +1,328 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+import * as React from 'react'
+import Enum from '../../react-component/container/input-wrappers/enum'
+import Text from '../../react-component/container/input-wrappers/text'
+import fetch from '../../react-component/utils/fetch'
+
+const _ = require('underscore')
+const user = require('component/singletons/user-instance')
+const common = require('js/Common')
+const announcement = require('component/announcement')
+
+const deepCopy = function(o) {
+  return JSON.parse(JSON.stringify(o))
+}
+
+const getGroups = function(groups = [], groupsRead = []) {
+  // only display the roles the current user has (even if other roles have
+  // permissions on this metacard)
+  return user
+    .get('user')
+    .get('roles')
+    .map(role => {
+      return {
+        id: common.generateUUID(),
+        type: 'role',
+        access: groups.includes(role)
+          ? 'write'
+          : groupsRead.includes(role)
+            ? 'read'
+            : 'none',
+        value: role,
+      }
+    })
+}
+
+const getIndividuals = function(
+  owner,
+  individuals = [],
+  individualsRead = [],
+  accessAdministrators = []
+) {
+  return _.union(individuals, individualsRead, accessAdministrators)
+    .filter(email => email !== owner) // hide owner
+    .map(email => {
+      return {
+        id: common.generateUUID(),
+        type: 'user',
+        access: accessAdministrators.includes(email)
+          ? 'share'
+          : individuals.includes(email)
+            ? 'write'
+            : individualsRead.includes(email)
+              ? 'read'
+              : 'none',
+        value: email,
+      }
+    })
+}
+
+export default class WorkspaceSharing extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      items: [],
+    }
+  }
+
+  componentDidMount() {
+    fetch(`/search/catalog/internal/metacard/${this.props.id}`)
+      .then(response => response.json())
+      .then(data => {
+        const metacard = data.metacards[0]
+        const items = getGroups(
```

Comment:
❓ 	Is the ordering of the items consistent? Should we sort them alphabetically by value?